src/commands/scanner/run.ts

```diff
@@ -181,16 +176,15 @@ export default class Run extends ScannerCommand {
 			// Default properly, again, as we did earlier.
 			const format: OUTPUT_FORMAT = this.flags.format || OUTPUT_FORMAT.TABLE;
 			// If we're just supposed to dump the output to the console, what precisely we do depends on the format.
-			if (format === OUTPUT_FORMAT.CSV) {
+			if (format === OUTPUT_FORMAT.CSV && typeof output === 'string') {
 				// The CSV is just one giant string that we can dump directly to the console.
 				this.ux.log(output);
-			} else if (format === OUTPUT_FORMAT.XML || format === OUTPUT_FORMAT.JUNIT) {
+			} else if ((format === OUTPUT_FORMAT.XML || format === OUTPUT_FORMAT.JUNIT) && typeof output === 'string') {
 				// For XML, we can just dump it to the console.
 				this.ux.log(output);
-			} else if (format === OUTPUT_FORMAT.TABLE && output.length > 0) {
+			} else if (format === OUTPUT_FORMAT.TABLE && typeof output === 'object') {
 				// For tables, don't even bother printing anything unless we have something to print.
-				const outputObj = JSON.parse(output);
-				this.ux.table(outputObj.rows, outputObj.columns);
+				this.ux.table(output.rows, output.columns);
```

Comment:
For some reason, the addition of type checking to the conditions makes me wonder if we should have an `else` at the end, just in case.